app/src/main/java/space/taran/arknavigator/mvp/model/repo/index/ResourcesIndex.kt

```diff
@@ -20,7 +20,7 @@ interface ResourcesIndex {
 
     suspend fun getMeta(id: ResourceId): ResourceMeta
 
-    suspend fun reindex()
+    suspend fun reindex(indexFailedPathCallback: IndexFailedPathCallback)
```

Comment:
I think the callback should be replaced with 
`val kindDetectFailedFlow: SharedFlow<Path>`

In `AggregatedResourcesIndex` you just need to merge all the flows from the shards

Also in the presenters in the `onFirstViewAttach` you just need to subscribe to this flow and show toasts

Reply:
I have implemented `SharedFlow` as you suggested and show `toasts` for each fail:
The problem : When multiple failed occurred, the last `toast` will be displayed. others' toasts are quickly disappeared.

Reply:
It's okay, we'll replace the regular toast with a custom one

Reply:
ok